src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -995,33 +1007,42 @@ protected override void EndProcessing()
                 }
             }
 
+            bool hasTimedOut = false;
             if (_numberOfProcessesToWaitFor > 0)
             {
                 if (_timeOutSpecified)
                 {
-                    _waitHandle.WaitOne(_timeout * 1000);
+                    hasTimedOut = !_waitHandle.WaitOne(_timeout * 1000);
                 }
                 else
                 {
                     _waitHandle.WaitOne();
                 }
             }
 
-            foreach (Process process in _processList)
+            if (hasTimedOut)
```

Comment:
We should preserve old behavior for "wait all processes":
```suggestion
            if (hasTimedOut || (!Any && _numberOfProcessesToWaitFor > 0))
```